apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/valueset/ValueSetCreator.java

```diff
@@ -110,64 +94,14 @@ public long getCodeSetGroupID() {
 		return maxGroupID;
 	}
 
-	public ValueSetCommand.AddValueSet asAdd(final ValueSetRequest request, long userId) {
+	public ValueSetCommand.AddValueSet asAdd(final ValueSetCreateRequest request, long userId) {
 		return new ValueSetCommand.AddValueSet(
-				request.getAssigningAuthorityCd(),
-				request.getAssigningAuthorityDescTxt(), 
-				request.getCodeSetDescTxt(), 
-				request.getEffectiveFromTime() != null ? request.getEffectiveFromTime() : Instant.now() ,
-				request.getEffectiveToTime(), 
-				request.getIsModifiableInd(), 
-				request.getNbsUid(),
-				request.getSourceVersionTxt(), 
-				request.getSourceDomainNm(), 
-				"A",
-				request.getStatusToTime(), 
-				request.getCodeSetGroup(), 
-				request.getAdminComments(),
-				request.getValueSetNm().toUpperCase(), 
-				request.getLdfPicklistIndCd(), 
-				request.getValueSetCode(),
-				request.getValueSetTypeCd(), 
-				request.getValueSetOid(), 
-				request.getValueSetStatusCd(),
-				request.getValueSetStatusTime(),
-				request.getParentIsCd(), 
-				Instant.now(), 
+				request.valueSetType(),
```

Comment:
Suggestion: The valid types in Classic are determined by the following:
```java
codeSql = "select code \"key\", code_desc_txt \"value\" FROM " + NEDSSConstants.SYSTEM_REFERENCE_TABLE + "..code_value_general where code_set_nm='NBS_QUESTION_TYPE' and code<>'SYS'";            
```
source: `SRTMapDAOImpl.java #3908`

Perhaps we could verify that the provided `valueSetType` matches a code found in this value set?

Reply:
I found a CodeSet enum that we can use.